src/lib/run_environment.ml

```diff
@@ -94,10 +94,37 @@ module Tool = struct
   end
 end
 
+module Make_fun = struct
+  module Requirement = struct
+    type t = [
+      | `Processors of int
+      | `Internet_access
+      | `Memory of [ `GB of float | `Decent | `Big ]
+      | `Quick_run
+      | `Spark of string list (* custom parameters *)
+      | `Custom of string
+    ] [@@deriving yojson, show]
+  end
```

Comment:
I'm as much a fan of regular variants as the next guy, but keeping this open means consumers of Biokepi can add their own tools with their own requirements without having to modify Biokepi, right?